doc/ug/getting_started_hw_design.md

```diff
@@ -90,3 +90,42 @@ To do so, follow the steps below.
 * Open a Pull Request with the necessary code changes.
 
 If it is not clear on how to proceed, feel free to file an issue requesting assistance.
+
+## Change of Interrupts
+
+Change of the interrupts in a HWIP needs lots of extra works.
+DV environment, if exists, should have the newly introduced, or renamed ports in its TB module.
+TOP module, if the module has been instantiated, should be re-generated to have new or renamed ports.
+DIF SW should be regenerated too.
+The PLIC test should be updated also.
+
+### Update TOP
+
+```console
+$ make -C hw top
+```
+
+After you revise the IP `.hjson`, IP top module, IP register interface, and DV testbench `tb.sv`, you can re-generate top earlgrey module with above command.
+It reads the `.hjson` file and update the interrupt signal and re-generate PLIC module if needed.
+
+### New DIF
+
+Unfortunately, re-generating TOP does not resolve everything.
+If the interrupt name is changed or new interrupt is introduced, the DIF for the IP should be re-generated.
+
+```console
+$ ./util/make_new_dif.py --mode=regen --only=autogen
+```
```

Comment:
Could we just update the [`hw/Makefile` ](https://github.com/lowRISC/opentitan/blob/master/hw/Makefile) to call this too? Then we don't need this extra step.

Reply:
yeah i tend to agree...
@timothytrippel what do you think?

Reply:
oops, sorry i thought @eunchan left this comment haha.
But I totally agree we should do this.

Reply:
Google has its own style guide for clang-format mandated. Is there a simple way to avoid it?

Reply:
Are you referring to the fact that `util/make_new_dif.py` invokes clang-format and by calling `util/make_new_dif.py` in the `hw/Makefile` we will a similar problem that was addressed in #8909? If so, I have an idea how we can avoid the clang-format issues, but we should probable address that in another PR. So maybe for now, let's merge this the way it is, and once merged, I will file a separate issue about invoking `util/make_new_dif.py` from  `hw/Makefile`, and then I can implement the PR to address this and fix the clang-format issues. Does that sound good?

Reply:
Captured in #9650.